src/Rest/Routes/NewsRoute.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * The class register route for $className endpoint
+ *
+ * @package Unicorns\Rest\Routes
+ */
+
+declare(strict_types=1);
+
+namespace Unicorns\Rest\Routes;
+
+use Unicorns\Config\Config;
+use UnicornsVendor\EightshiftLibs\Helpers\Components;
+use UnicornsVendor\EightshiftLibs\Rest\Routes\AbstractRoute;
+use UnicornsVendor\EightshiftLibs\Rest\CallableRouteInterface;
+
+/**
+ * Class NewsRoute
+ */
+class NewsRoute extends AbstractRoute implements CallableRouteInterface
+{
+
+	/**
+	 * Method that returns project Route namespace.
+	 *
+	 * @return string Project namespace UnicornsVendor\for REST route.
+	 */
+	protected function getNamespace(): string
+	{
+		return Config::getProjectRoutesNamespace();
+	}
+
+	/**
+	 * Method that returns project route version.
+	 *
+	 * @return string Route version as a string.
+	 */
+	protected function getVersion(): string
+	{
+		return Config::getProjectRoutesVersion();
+	}
+
+	/**
+	 * Get the base url of the route
+	 *
+	 * @return string The base URL for route you are adding.
+	 */
+	protected function getRouteName(): string
+	{
+		return '/news';
+	}
+
+	/**
+	 * Get callback arguments array
+	 *
+	 * @return array Either an array of options for the endpoint, or an array of arrays for multiple methods.
+	 */
+	protected function getCallbackArguments(): array
+	{
+		return [
+			'methods' => static::READABLE,
+			'callback' => [$this, 'routeCallback'],
+			'permission_callback' => '__return_true'
+		];
+	}
+
+	/**
+	 * Method that returns rest response
+	 *
+	 * @param \WP_REST_Request $request Data got from endpoint url.
+	 *
+	 * @return \WP_REST_Response|mixed If response generated an error, WP_Error, if response
+	 *                                is already an instance, WP_HTTP_Response, otherwise
+	 *                                returns a new WP_REST_Response instance.
+	 */
+	public function routeCallback(\WP_REST_Request $data)
+	{
+		$params = $data->get_params();
+
+		$returnHtml = false;
+
+		foreach ($params as $name => $value) {
```

Comment:
you rely that you will always have $params available? If they are not you will get and fatal error. Please write additional check here

Reply:
Wow, I just realized how lucky I was with this.
`get_params` returns an empty array if no parameters are available it wouldn't cause any errors.

But I still added a `if (isset(` check :)

Reply:
Yeah, you don't need an extra check if the default value is an empty value. Foreach will skip it. Just make sure that you don't have any potential non defined values.

In these cases, PHPStan is a lifesaver 🙂